crates/core/tedge_agent/src/agent.rs

```diff
@@ -225,19 +225,11 @@ impl SmAgent {
         })
     }
 
-    pub fn log_dir(&self) -> PathBuf {
-        self.config.log_dir.clone()
-    }
-
     #[instrument(skip(self), name = "sm-agent")]
-    pub async fn init(
-        &mut self,
-        config_dir: PathBuf,
-        log_path: PathBuf,
-    ) -> Result<(), anyhow::Error> {
+    pub async fn init(&mut self, config_dir: PathBuf) -> Result<(), anyhow::Error> {
         // `config_dir` by default is `/etc/tedge` (or whatever the user sets with --config-dir)
-        // `log_path` is by default `/var/log` (or whatever the user sets with tedge config set logs.path)
         let config_dir = config_dir.display();
+        let log_path = self.config.log_dir.clone();
         let log_path = log_path.display();
```

Comment:
```suggestion
        let log_path = self.config.log_path.display();
```

I think the `clone()` call can be removed here, right?

Reply:
thanks :D